extensions/crossmodel-lang/src/glsp-server/mapping-diagram/handler/create-edge-operation-handler.ts

```diff
@@ -40,11 +40,12 @@ export class MappingEdgeCreationOperationHandler extends JsonCreateEdgeOperation
          const container = this.modelState.mapping.target;
          const sourceElement = this.modelState.index.findSemanticElement(sourceElementId, isSourceObjectAttribute);
          const targetElement = this.modelState.index.findSemanticElement(targetElementId, isTargetObjectAttribute);
-         if (!targetElement || !sourceElement) {
+         const sourceEntity = <LogicalEntity | undefined>getOwner(sourceElement);
+         if (!targetElement || !targetElement.id || !sourceElement || !sourceElement.id || !sourceEntity || !sourceEntity.id) {
```

Comment:
nit: Could be shortened to:
```suggestion
         if (!targetElement?.id || !sourceElement?.id || !sourceEntity?.id) {
```